lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts

```
@@ -681,6 +682,27 @@ describe("Acquires a token successfully via an IMDS Managed Identity", () => {
             ).toBe(false);
         }, 10000); // double the timeout value for this test because it waits two seconds in between the acquireToken call and the cache lookup
 
+        test("ensures an ISO 8601 date returned by the Managed Identity is converted to a Unix timestamp (seconds since epoch)", async () => {
```

can you extend the tests to cover few more cases, see MSAL NET test coverage [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dec281197a8f49f343263247dbff3c3090185736/tests/Microsoft.Identity.Test.Unit/UtilTests/DateTimeHelperTests.cs)

**MSAL .NET tests validate:**

- **Unix timestamps** (seconds since epoch) for past, present, and future values.
- **ISO 8601 format** (e.g., 2024-10-18T19:51:37.0000000+00:00).
- **Common date formats** (e.g., MM/dd/yyyy HH:mm:ss, yyyy-MM-dd HH:mm:ss).
- **Invalid formats** (which should throw an exception).

**also consider testing:**

- **Time zone variations** Example: 2024-10-18T19:51:37-07:00 (time zones with offsets).
- **Negative timestamps** Example: -1697490590 (handling of past timestamps).
- **Extreme future timestamps** Example: 31556995200 (1000 years into the future).
- **Fractional Unix timestamps** Example: 1697490590.123 (valid timestamps with milliseconds).

---

Unix timestamps and their test coverage already exist. The purpose of this PR is to add functionality for ISO 8601 timestamps.

We can open a new PR if you'd like to see additional test coverage.

---

Can you add tests for ISO 8601 edge cases like time zones (+05:30, -07:00), fractional seconds (.123Z), date-only formats (2024-10-18), and invalid values (25:00:00)?

---

here is the data @Robbie-Microsoft 

```js
// Example test data for ISO 8601 variants:
const testDates = [
  {
    input: "2025-01-14T19:51:37-07:00",
    description: "Valid ISO 8601 with timezone offset"
  },
  {
    input: "2025-01-14T19:51:37.123Z",
    description: "Valid ISO 8601 in UTC with fractional seconds"
  },
  {
    input: "2025-01-14T25:00:00Z",
    description: "Invalid time (25 is not a valid hour)"
  },
  {
    input: "2025-01-14T19:51:37Z",
    description: "Valid ISO 8601 in UTC with no fractional seconds"
  },
  {
    input: "2025-01-14",
    description: "Date-only (no time)"
  },
];
```

---

Comment not resolved?